### PR TITLE
Feature/cache update

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/cache/ProjectCacheSynchronizer.java [hunk 1]
```diff
@@ -63,22 +63,48 @@ public ProjectCacheSynchronizer(QualityProfileLoader qualityProfileLoader, Proje
...
+      c1.get(Calendar.YEAR) == c2.get(Calendar.YEAR);
+  }
+
+  private static boolean shouldUpdate(Date lastUpdate) {
+    return !isToday(lastUpdate);
+  }
+
   public void load(String projectKey, boolean force) {
     Date lastSync = cacheStatus.getSyncStatus();
+    boolean failOnError = true;
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/cache/ProjectCacheSynchronizer.java]: MAJOR Remove this useless assignment to local variable "failOnError". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1854)

File: sonar-batch/src/main/java/org/sonar/batch/cache/ProjectCacheSynchronizer.java [hunk 2]
```diff
@@ -63,22 +63,48 @@ public ProjectCacheSynchronizer(QualityProfileLoader qualityProfileLoader, Proje
...
 
     if (lastSync != null) {
-      if (!force) {
+      if (force) {
+        LOG.info("-- Found project [{}] cache [{}], synchronizing data (forced)..", projectKey, lastSync);
+      } else if (shouldUpdate(lastSync)) {
+        LOG.info("-- Found project [{}] cache [{}], synchronizing data..", projectKey, lastSync);
+        failOnError = false;
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/cache/ProjectCacheSynchronizer.java]: MAJOR Remove this useless assignment to local variable "failOnError". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1854)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
LGTM apart SonarQube issues

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**